commit 9283ef60c5aa5c8700300453d7826ae3fa8acebd
Author: bertin <macportsuser@macports.org>
Date:   Fri May 23 00:02:07 2025 +0200

    /opt/local/site-ports/sysutils/MacPorts/files/patch-export-logfilename.diff

diff --git a/src/macports1.0/macports.tcl b/src/macports1.0/macports.tcl
index 9db20a226..08c52b70c 100644
--- a/src/macports1.0/macports.tcl
+++ b/src/macports1.0/macports.tcl
@@ -2140,6 +2140,9 @@ proc macports::worker_init {workername portpath porturl portbuildpath options va
     $workername alias _mportkey _mportkey
     $workername alias _mport_archs macports::_mport_archs
     $workername alias _mport_supports_archs macports::_mport_supports_archs
+    ## RJVB
+    $workername alias get_logfile macports::get_logfile
+    $workername alias flush_logfile macports::flush_logfile
 
     # New Registry/Receipts stuff
     $workername alias registry_new registry::new_entry
@@ -6917,6 +6920,24 @@ proc macports::shellescape {arg} {
     return [regsub -all -- {[^A-Za-z0-9.:@%/+=_-]} $arg {\\&}]
 }
 
+## RJVB
+# return the logfile name, or an empty string
+proc macports::get_logfile {} {
+	if {[info exists ::debuglogname]} {
+		return $::debuglogname
+	} else {
+		return ""
+	}
+}
+proc macports::flush_logfile {} {
+	if {[info exists ::debuglog]} {
+		flush $::debuglog
+		return yes
+	} else {
+		return no
+	}
+}
+
 ##
 # Given a list of maintainers as recorded in a Portfile, return a list of lists
 # in [key value ...] format describing all maintainers. Valid keys are 'email'
